src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/CollectionRuleOptionsTests.cs

```diff
@@ -829,6 +829,156 @@ public Task CollectionRuleOptions_LoadProfilerAction_ClsidPropertyValidation()
                 });
         }
 
+        [Fact]
+        public Task CollectionRuleOptions_SetEnvironmentVariable_MinimumOptions()
+        {
+            const string VariableName = "MyProfiler_OptionA";
+
+            return ValidateSuccess(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddSetEnvironmentVariableAction(VariableName);
+                },
+                ruleOptions =>
+                {
+                    Assert.Single(ruleOptions.Actions);
+                    ruleOptions.VerifySetEnvironmentVariableAction(0, VariableName, null);
+                });
+        }
+
+        [Fact]
+        public Task CollectionRuleOptions_SetEnvironmentVariable_RoundTrip()
+        {
+            const string VariableName = "MyProfiler_OptionA";
+            const string VariableValue = "MyValue!@#$%^&*()_+-{}{]|";
+
+            return ValidateSuccess(
+                rootOptions =>
+                {
+                    rootOptions.CreateCollectionRule(DefaultRuleName)
+                        .SetStartupTrigger()
+                        .AddSetEnvironmentVariableAction(VariableName, VariableValue);
+                },
+                ruleOptions =>
+                {
+                    Assert.Single(ruleOptions.Actions);
+                    ruleOptions.VerifySetEnvironmentVariableAction(0, VariableName, VariableValue);
+                });
+        }
+
+        [Fact]
+        public async Task CollectionRuleOptions_SetEnvironmentVariable_NamePropertyValidation()
+        {
+            await ValidateFailure(
```

Comment:
Since this looks like the exact same validation three times in a row (just with a different illegal environment variable name), should we consider converting this to theory and just passing the illegal name as a parameter?

Reply:
I had it does this way to begin with but since the action within `ValidateFailure(...)` is so fast, I didn't want to add the extra overhead of extra test method launches, since we pay a penalty for each test case.